java/simulator/src/main/java/com/hazelcast/simulator/coordinator/PerformanceStatsCollector.java

```diff
@@ -71,28 +71,38 @@ public String formatIntervalPerformanceNumbers(String testId) {
             return "";
         }
 
-        String latencyUnit = "µs";
-        long latencyAvg = NANOSECONDS.toMicros(round(latest.getIntervalLatencyAvgNanos()));
-        long latency999Percentile = NANOSECONDS.toMicros(latest.getIntervalLatency999PercentileNanos());
-        long latencyMax = NANOSECONDS.toMicros(latest.getIntervalLatencyMaxNanos());
-
-        if (latencyAvg > DISPLAY_LATENCY_AS_MICROS_MAX_VALUE) {
-            latencyUnit = "ms";
-            latencyAvg = MICROSECONDS.toMillis(latencyAvg);
-            latency999Percentile = MICROSECONDS.toMillis(latency999Percentile);
-            latencyMax = MICROSECONDS.toMillis(latencyMax);
-        }
+        double latencyAvgNs = latest.getIntervalLatencyAvgNanos();
+        double latency999PercentileNs = latest.getIntervalLatency999PercentileNanos();
+        double latencyMaxNs = latest.getIntervalLatencyMaxNanos();
 
         return format("%s ops %s ops/s %s %s (avg) %s %s (%sth) %s %s (max)",
                 formatLong(latest.getOperationCount(), OPERATION_COUNT_FORMAT_LENGTH),
                 formatDouble(latest.getIntervalThroughput(), THROUGHPUT_FORMAT_LENGTH),
-                formatLong(latencyAvg, LATENCY_FORMAT_LENGTH),
-                latencyUnit,
-                formatLong(latency999Percentile, LATENCY_FORMAT_LENGTH),
-                latencyUnit,
+                formatLong(toPrettyValue(latencyAvgNs), LATENCY_FORMAT_LENGTH),
+                toPrettyUnit(latencyAvgNs),
+                formatLong(toPrettyValue(latency999PercentileNs), LATENCY_FORMAT_LENGTH),
+                toPrettyUnit(latency999PercentileNs),
                 INTERVAL_LATENCY_PERCENTILE,
-                formatLong(latencyMax, LATENCY_FORMAT_LENGTH),
-                latencyUnit);
+                formatLong(toPrettyValue(latencyMaxNs), LATENCY_FORMAT_LENGTH),
+                toPrettyUnit(latencyMaxNs));
+    }
+
+    private static long toPrettyValue(double valueNs) {
```

Comment:
I think it would be helpful to document the unit of the returned value (nanoseconds?) to avoid someone later getting the scale wrong.
Ideally returning a `Duration` object to make this impossible.

Reply:
I have added the documentation.